specs/proposals/diagnostic-experience.md

```diff
@@ -0,0 +1,249 @@
+# Improve Notation Diagnostic Experience
+
+> [!NOTE]
+> The version of this specification is for `notation` v2.0.0 Beta.1. It is subject to change until v2.0.0 is released. 
+
+## Overview
+
+Notation v1.x offers two global options, `--verbose` and `--debug`, which allows users to generate verbose output and debug logs respectively. These features facilitate both users and developers in inspecting `notation`'s performance, interactions with external services and internal systems, and in diagnosing issues by providing a clear picture of the tool's operations.
+
+This proposal document aims to:
+
+1. Identify the usability issues of the `--verbose` and `--debug` options.
+2. Clarify the concepts of different types of output and logs for diagnostic purposes.
+3. List the guiding principles to write comprehensive, clear, and conducive debug output and debug logs for effective diagnosis.
+4. Propose solutions to improve the diagnostic experience for `notation` CLI users and developers.
+
+## Problem Statement
+
+Given the diverse roles and scenarios in which `notation` CLI is utilized, we have received feedback from users and developers to improve the diagnostic experience as described in [GitHub issue #1247](https://github.com/notaryproject/notation/issues/1247). 
+
+Specifically, there are known issues or confusion when diagnosing a problem.
+
+- The user is confused about when to use `--verbose` and `--debug`. 
+- Both `--verbose` and `--debug` output the INFO level logs which is duplicated information. 
+- Poor readability of debug logs. No separator lines between request and response information.
+- Timestamp does not use [Nanoseconds](https://pkg.go.dev/time#Duration.Nanoseconds) precision, which is not accurate to trace historical operation.
+- No easy way to get the user environment information. This causes a higher cost to reproduce the issues for `notation` developers.
+
+## Scenarios
+
+Alice is a DevSecOps engineer who uses `notation` CLI to sign artifacts in a CI/CD pipeline. The signing step failed for some reasons with debug logs generated. Alice tries to figure out the root cause by reading the debug logs but she can't locate the problem. In this scenario, she has to submit a GitHub issue to request help from the Notary Project community and provide debug logs. The debug logs will be used by `notation` developers to analyze the operations and locate the root cause. `notation` developers need to reproduce the failed steps for troubleshooting. 
```

Comment:
The scenario is not accurate for addressing the problem. Because currently there are debug logs, but there are issues as you listed, those should be the problem that Alice faces, and Alice solves it by doing what.

Reply:
I fulfilled the scenario description by referencing the issues from the problem statement section.